data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/util/UserMetadataUpdater.scala

```diff
@@ -0,0 +1,254 @@
+package org.sunbird.dp.usercache.util
+
+import java.util
+import java.util.Collections
+
+import com.datastax.driver.core.Row
+import com.datastax.driver.core.querybuilder.{Clause, QueryBuilder}
+import org.slf4j.LoggerFactory
+import org.sunbird.dp.core.cache.{DataCache, RedisConnect}
+import org.sunbird.dp.core.job.Metrics
+import org.sunbird.dp.core.util.CassandraUtil
+import org.sunbird.dp.usercache.domain.Event
+import org.sunbird.dp.usercache.task.UserCacheUpdaterConfigV2
+
+import scala.collection.JavaConverters.mapAsJavaMap
+import scala.collection.mutable
+
+class UserMetadataUpdater (config: UserCacheUpdaterConfigV2) {
+
+  private[this] val logger = LoggerFactory.getLogger(classOf[UserMetadataUpdater])
+  private val dataCache: DataCache = new DataCache(config, new RedisConnect(config.metaRedisHost, config.metaRedisPort, config), config.userStore, config.userFields)
+  private val cassandraConnect: CassandraUtil = new CassandraUtil(config.cassandraHost, config.cassandraPort)
+  private val custodianRootOrgId: String = getCustodianRootOrgId()
+  dataCache.init()
+
+  def userUpdater(event: Event, metrics: Metrics): Unit = {
+    metrics.incCounter(config.totalEventsCount)
+    Option(event.getId).map(id => {
+      Option(event.getState).map(name => {
+        val userData: mutable.Map[String, AnyRef] = name.toUpperCase match {
+          case "CREATE" | "CREATED" => createAction(id, event, metrics)
+          case "UPDATE" | "UPDATED" => updateAction(id, event, metrics)
+          case _ => {
+            logger.info(s"Invalid event state name either it should be(Create/Created/Update/Updated) but found $name for ${event.mid()}")
+            metrics.incCounter(config.skipCount)
+            mutable.Map[String, AnyRef]()
+          }
+        }
+        if (!userData.isEmpty) {
+          val redisData = userData.filter{f => (null != f._2)}
+          val data = redisData.map(f => (f._1, f._2.toString))
```

Comment:
toString is risky. can we have a stringify() method and properly encode for each data type? For ex: List & Map types should be encoded as JSON

Reply:
Done